client/my-sites/pages/page-list.jsx

```diff
@@ -202,7 +213,7 @@ var Pages = React.createClass( {
 			rows = [];
 
 		// pages have loaded, sites have loaded, and we have a site instance or are viewing all-sites
-		if ( pages.length && this.props.sites.initialized ) {
+		if ( pages.length && this.props.hasSites ) {
```

Comment:
@mcsf and I chatted in DM to get to the bottom of this, seems like it was happening only on the "trashed pages" route & when localStorage was cleared but not IndexedDB.

I went ahead and created a new global state value to indicate when the ui had performed site selection (including setting "all sites") in b6d4b08.

Since I'm not sure what it would involve to refactor the [`selectedSiteId` key](https://github.com/Automattic/wp-calypso/blob/61d0b5ff0f33d6b1edeb7421a92dca7a1a79ae6c/client/state/ui/reducer.js#L32-L46) to be `false` until a `SELECTED_SITE_SET` event is dispatched, this seemed like a pretty low-impact workaround.